src/Attribute/Parameter/Di.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Hydrator\Attribute\Parameter;
+
+use Attribute;
+use Yiisoft\Hydrator\ParameterAttributeInterface;
+
+#[Attribute(Attribute::TARGET_PROPERTY | Attribute::TARGET_PARAMETER)]
+final class Di implements ParameterAttributeInterface
```

Comment:
Yes. Set specific object ID for object constructor parameter:

```php
 __construct(#[Di('CarDecorator')] CarInterface $car)
 ```

Reply:
https://github.com/yiisoft/di/issues/307 😃

Reply:
For models it's actual. Has other cases?

Reply:
I think, every case if container has string definitions. For example: ability to autowire `connection2` definition in repository/service class instead of do `$container->get('connection2')`.
But I'm not sure it's good idea.

Reply:
It's should not configure in service, because break dependency inversion principle.